pymysqlreplication/row_event.py

```diff
@@ -79,8 +79,9 @@ def __init__(self, from_packet, event_size, table_map, ctl_connection, **kwargs)
         #Body
         self.number_of_columns = self.packet.read_length_coded_binary()
         self.columns = self.table_map[self.table_id].columns
+        column_schemas = self.table_map[self.table_id].column_schemas
 
-        if len(self.columns) == 0:  # could not read the table metadata, probably already dropped
+        if len(column_schemas) == 0:  # could not read the table metadata, probably already dropped
```

Comment:
if table dropped, we can notice from column_schemas info

self.column_schemas = self._ctl_connection._get_table_information(self.schema, self.table)
self.table_obj = Table(self.column_schemas, self.table_id, self.schema,self.table, self.columns)